src/libraries/Common/src/Interop/Windows/Netapi32/Interop.DsGetDcNext.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Netapi32
+    {
+        [LibraryImport(Libraries.Netapi32, EntryPoint = "DsGetDcNextW")]
+        internal static partial int DsGetDcNext(
+            IntPtr getDcContextHandle,
+            ref IntPtr sockAddressCount,
```

Comment:
```suggestion
            out IntPtr sockAddressCount,
```

Reply:
Done in 80536c19b1c85a2a4606868670b9016688500468, however I noticed that the [documentation](https://learn.microsoft.com/en-us/windows/win32/api/dsgetdc/nf-dsgetdc-dsgetdcnextw) states that the caller should be calling `LocalFree` on the memory pointed to by `sockAddresses` but we don't seem to do that in https://github.com/dotnet/runtime/blob/920cd6e2b01d0fbede8882d015329e9b4d448c98/src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs#L30

Is this something that should be fixed with a call to `Interop.Kernel32.LocalFree` (or `Marshal.FreeHGlobal`?) after each call (similar to how `dnsHostNamePtr` memory _is_ freed, if necessary, with `NetApiBufferFree`), or am I misunderstanding something?

Reply:
If it helps explain what I mean, it looks like the sock structures here are accumulating with each call to `DsGetDcNext` and remain past the method returning - if I put a call to `FreeHGlobal(sockAddressList)` in the `finally` block where we free the `dcDnsHostNamePtr` then the memory appears to get re-used across calls...

<img width="1221" alt="image" src="https://github.com/dotnet/runtime/assets/5628773/8e5fa5bb-abdc-4216-8c4e-a4c2d2e74be4">

Reply:
> however I noticed that the [documentation](https://learn.microsoft.com/en-us/windows/win32/api/dsgetdc/nf-dsgetdc-dsgetdcnextw) states that the caller should be calling LocalFree on the memory pointed to by sockAddresses but we don't seem to do that

Looks like you found a leak :) Nice job! Yes, this should be cleaned up prior to moving to the next iteration.

Reply:
Sweet, fixed in 4f9d43742b52d994edd4278f3490e3d5431d5310